contract/src/test/kotlin/io/provenance/scope/loan/contracts/AppendLoanDocumentsContractUnitTest.kt

```diff
@@ -166,13 +177,79 @@ class AppendLoanDocumentsContractUnitTest : WordSpec({
                     }
                     AppendLoanDocumentsContract(
                         existingDocs = randomExistingDocuments.toRecord(),
+                        existingServicingData = ServicingData.getDefaultInstance(), // Unused
                     ).appendDocuments(
                         newDocs = randomNewDocuments.toRecord()
                     )
                 }
             }
         }
     }
+    "appendServicingDocuments" When {
```

Comment:
can we have a happy path case here?

Reply:
Added a test case with some complex Arb construction which should handle all possible happy path scenarios. Let me know if you see any cases it doesn't cover, mistakes, or better ways to write the code there.